frontend/src/views/CellCards/components/CellCard/components/OntologyDagView/components/AnimatedNodes/index.tsx

```diff
@@ -0,0 +1,149 @@
+              n_cells_rollup: datum.n_cells_rollup,
+            },
+          });
+        }
+      }
+    }
+  };
+
+  return (
+    <NodeGroup
+      data={tree.descendants()}
+      keyAccessor={(d: HierarchyPointNode<TreeNodeWithState>) => d.data.id}
+      start={(node: HierarchyPointNode<TreeNodeWithState>) => {
+        return node.parent
+          ? {
+              top: node.parent.data.x0 ?? node.parent.x,
+              left: node.parent.data.y0 ?? node.parent.y,
+              opacity: 0,
+              timing: { duration },
+            }
+          : {
+              top: node.x,
+              left: node.y,
+              opacity: 0,
+              timing: { duration },
+            };
+      }}
+      enter={(node: HierarchyPointNode<TreeNodeWithState>) => {
+        return {
+          top: [node.x],
+          left: [node.y],
+          opacity: [1],
+          timing: { duration },
+        };
+      }}
+      update={(node: HierarchyPointNode<TreeNodeWithState>) => {
+        return {
+          top: [node.x],
+          left: [node.y],
+          opacity: [1],
+          timing: { duration },
+        };
+      }}
+      leave={() => {
+        return { opacity: [0], timing: { duration: 0 } };
+      }}
+    >
+      {(nodes) => {
+        return (
+          <Group>
+            {nodes.map(({ key, data: node, state }) => {
+              const isInCorpus =
+                (node.data.id.split("__").at(0)?.replace("_", ":") ?? "") in
+                cellTypesById;
+              return (
+                <Node
+                  key={`${key}-node`}
+                  isInCorpus={isInCorpus}
+                  animationKey={key}
+                  node={node}
+                  isTargetNode={node.data.id.split("__").at(0) === cellTypeId}
+                  handleMouseOver={handleMouseOver}
+                  handleMouseOut={hideTooltip}
+                  maxWidth={NODE_SPACINGS[1] - 20}
+                  left={state.left}
+                  top={state.top}
+                  opacity={state.opacity}
+                  handleClick={() => {
+                    if (duration === 0) {
+                      setDuration(TREE_ANIMATION_DURATION);
+                    }
+                    if (node.data.id.startsWith("dummy-child")) {
+                      if (node.parent) {
+                        node.parent.data.showAllChildren = true;
+                      }
+                    }
+                    node.data.isExpanded = !node.data.isExpanded;
+                    if (!node.data.isExpanded) {
+                      node.data.showAllChildren = false;
+                      collapseAllDescendants(node);
+                    }
+                    toggleTriggerRender();
+                  }}
+                />
+              );
+            })}
+          </Group>
+        );
+      }}
+    </NodeGroup>
+  );
+}
+
+function collapseAllDescendants(node: HierarchyPointNode<TreeNodeWithState>) {
```

Comment:
modify all descendants of a node to have `isExpanded=false`. This collapses them